source-checker.pl

```diff
@@ -200,7 +200,13 @@ BEGIN
     if (%patches) {
         # parsing changes
         for my $changes (@changes) {
-            my $diff = diff "$old/$changes", "$dir/$changes";
+            my $diff = "";
```

Comment:
The variable needs to be defined outside of condition scope otherwise the diff seems to see nothing.